api/views.py

```diff
@@ -139,6 +165,133 @@ def _run_job(self, job):
         print_file.delay(job.id)
         logger.info('User %s submitted job: "%s"', self.request.user.username)
         return job
+    
+    def _change_order(self, job, new_order):
+        for id in new_order:
+            if not job.artefacts.filter(id=id).exists():
+                raise ValueError(f"Artefact with id {id} does not exist in this job")
+        if len(new_order) != job.artefacts.count():
+            raise ValueError("New order list must contain all artefact IDs exactly once")
+        
+        idx = 1
+        for id in new_order:
+            artefact = job.artefacts.filter(id=id).first()
+            artefact.order = idx
+            artefact.save()
+            idx += 1
+        return job
+    
+    @action(detail=True, methods=['get'], name='Get artefacts')
+    def artefacts(self, request, pk=None):
+        job = self.get_object()
+        artefacts = job.artefacts.all()
+        artefact_data = [
+            {
+                "id": artefact.id,
+                "file_name": artefact.file.name,
+                "mime_type": artefact.mime_type,
+                "artefact_type": artefact.artefact_type,
+                "order": artefact.order
+            }
+            for artefact in artefacts
+        ]
+        return Response(artefact_data)
+
+    @action(detail=True, methods=['delete'], name='Delete artefact')
+    def delete_artefact(self, request, pk=None):
+        job = self.get_object()
+        serializer = DeleteJobArtefactRequestSerializer(data=request.query_params)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        artefact_id = serializer.validated_data['file_id']
+        try:
+            artefact = job.artefacts.get(id=artefact_id)
+            artefact.delete()
+            return Response({"message": f"Artefact {artefact_id} deleted successfully"}, status=status.HTTP_200_OK)
+        except JobArtefact.DoesNotExist:
+            return Response({"error": f"Artefact with id {artefact_id} does not exist"}, status=status.HTTP_404_NOT_FOUND)
+        
+    @action(detail=True, methods=['get'], name='Get properties')
+    def properties(self, request, pk=None):
```

Comment:
Consider returning the properties as a field in the `GET /api/jobs/123/` request